### PR TITLE
fix(perf): Add ability to feature flag search terms

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

There are several expensive inserts that we do that can have large cardinality. These inserts are due to needing to be able to search for the nested fields. However, I'd like to deprecate them. The agreed upon deprecation schedule is:

Release 1: Flag but keep as is
Release 2: Flag is on so no writes, but changeable
Release 3: _Can_ be removed

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Flip the flag back and forth. Will do a scale run with the flags on to see the impact on the scale and will post the results

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -41,6 +41,6 @@ func saveFeature(name, envVar string, defaultValue, overridable bool) FeatureFla
 		defaultValue: defaultValue,
 		overridable:  overridable,
 	}
-	Flags[f.Name()] = f
+	Flags[f.envVar] = f
```
Comment (theencee) [inline pkg/features/features.go:44]: wait was this always broken!?

Comment (connorgorman) [reply pkg/features/features.go:44]: Yeah, but we only ever iterated over it for the values

File: pkg/postgres/walker/walker.go [hunk 2]
```diff
@@ -275,7 +276,12 @@ func getPostgresOptions(tag string, topLevel bool, ignorePK, ignoreUnique, ignor
 		case strings.HasPrefix(field, "type"):
 			typeName := field[strings.Index(field, "(")+1 : strings.Index(field, ")")]
 			opts.ColumnType = typeName
-
+		case strings.HasPrefix(field, "flag"):
```
Comment (c-du) [inline pkg/postgres/walker/walker.go:279]: Should be "flag="? Also in consistent with the search case.

File: pkg/search/walker.go [hunk 3]
```diff
@@ -82,6 +84,17 @@ func (s *searchWalker) handleStruct(prefix string, original reflect.Type) {
...
+		if strings.HasPrefix(searchTag, "flag=") {
+			flag := stringutils.GetAfter(searchTag, "=")
+			ff, ok := features.Flags[flag]
+			if !ok {
+				log.Fatalf("flag %s is not a valid feature flag", flag)
+			}
+			if !ff.Enabled() {
+				continue
+			}
+			searchTag = ""
```
Comment (c-du) [inline pkg/search/walker.go:96]: That means we cannot combine the flat option with a valid search tag?

**Review by c-du — APPROVED** [review]

**Pull request merged**